src/app/search/items/search-filters/sockets.ts

```diff
@@ -312,28 +312,30 @@ const socketFilters: ItemFilterDefinition[] = [
     destinyVersion: 2,
     filter: () => (item) =>
       Boolean(
-        item.sockets?.allSockets.some(
-          (s) =>
-            s.plugged?.plugDef.plug.plugCategoryHash ===
-            PlugCategoryHashes.CraftingPlugsWeaponsModsEnhancers,
-        ),
+        (item.craftedInfo?.enhancementTier || 0) < 3 &&
+          item.sockets?.allSockets.some(
+            (s) =>
+              s.plugged?.plugDef.plug.plugCategoryHash ===
+              PlugCategoryHashes.CraftingPlugsWeaponsModsEnhancers,
+          ),
       ),
   },
   {
-    // this currently tests for full enhancedness, returning false for partially-enhanced items
     keywords: 'enhanced',
     description: tl('Filter.Enhanced'),
     destinyVersion: 2,
-    format: ['simple'], // TO-DO: add 'range' here
-    filter:
-      ({ lhs }) =>
-      (item) => {
-        if (lhs === 'is') {
-          // rules out partially-enhanced items
-          // the game explicitly warns you that partially-enhanced items stop looking masterworked
-          return item.crafted === 'enhanced' && item.masterwork;
-        }
-      },
+    format: ['simple', 'range'],
+    filter: ({ lhs, compare }) => {
```

Comment:
You might want to add an extra condition for whether the enhance plug exists - that way `enhanceable:0` would light up all items that are enhanceable but which have not been enhanced at all?

Reply:
I was debating whether to include non-enhanceable weapons in enhanced:0 (and not:enhanced) or not. Will start a discussion in Discord.